lib/syskit/network_generation/dataflow_dynamics.rb

```diff
@@ -618,9 +627,22 @@ def policy_for(
                           "#{sink_port_m.required_connection_type} " \
                           "on #{sink_port}"
                 end
+                policy
+            end
+
+            def policy_for(
+                source_task, source_port_name, sink_port_name, sink_task, fallback_policy
+            )
+                policy = {}
+                unless policy[:type]
```

Comment:
`policy` is the empty hash here. This will always be nil.

The `if` should obviously be checking the policy that is coming from the policy graph.

I would actually pass `policy` from the loop in `compute_policies_from` to the method, and add the merge_policy call at the end of this method instead of doing it in said loop.